### DISCUSSION TITLE
HardwareSerial::end

### DISCUSSION THREAD

**Discussion by ikkei3** [espressif/arduino-esp32#8627] [General]
What is the purpose of delay(10) in HardwareSerial::end()?
Is that a spell? Please remove it.

**Comment by SuGlider** [comment]
I found that this was inserted in the PR #5047 beacuse of the issue #5043
At this time I also see no purpose for this `delay(10)`.
I'll get it out.

**Comment by SuGlider** [comment]
This is fixed by #8634

**Comment by trickyau** [comment]
This revert seems to break HardwareSerial 0 on Seeeduino XIAO ESP32-C3. Downgrading back to 2.0.12 fixes it.

**Reply by SuGlider** [reply]
@trickyau - Seeeduino XIAO ESP32-C3 board seems to use the HWCDC as default Serial Communication. 
When I select the "XIAO_ESP32C3" board it has "USB CDC On Boot: Enabled" by default, indicating that `Serial` is actually the CDC port, not the HardwareSerial UART port.

I need more details about how the C3 was connected to the UART GPIOs and how to reproduce the issue. 
Please open an issue with all the information.
